src/main/java/hudson/plugins/emailext/EmailThrottler.java

```diff
@@ -0,0 +1,38 @@
+package hudson.plugins.emailext;
+
+public class EmailThrottler {
+    public static final int THROTTLING_LIMIT = 100; // Set your limit
+    private static EmailThrottler instance;
+    private int emailCount;
+    private long lastResetTime;
+    private static final long THROTTLING_PERIOD = 60 * 60 * 1000; // Set your period (in milliseconds)
+
+    public EmailThrottler() {
+        this.emailCount = 0;
+        this.lastResetTime = System.currentTimeMillis();
+    }
+
+    public static synchronized EmailThrottler getInstance() {
+        if (instance == null) {
+            instance = new EmailThrottler();
+        }
+        return instance;
+    }
+
+    public synchronized boolean isThrottlingLimitExceeded() {
+        long currentTime = System.currentTimeMillis();
+        if (currentTime - lastResetTime > THROTTLING_PERIOD) {
+            resetEmailCount();
+        }
```

Comment:
The returned value of `System.currentTimeMillis` is not guaranteed to be monotonically increasing. Prefer `System.nanoTime` for this purpose instead.

Reply:
Thank you! I have changed it in d894dd2